src/MongoDB.Driver/Encryption/MongocryptdFactory.cs

```diff
@@ -119,10 +119,10 @@ private bool ShouldMongocryptdBeSpawned(out string path, out string args)
                     path = string.Empty; // look at the PATH env variable
                 }
 
-                if (Directory.Exists(path))
+                if (string.IsNullOrEmpty(path) || Directory.Exists(path))
```

Comment:
nit: string.IsNullOrWhiteSpace?

Reply:
I thought about this, but I'm not sure how we need to handle the case when a user provides just spaces. Mostlikely it will be his mistake since he intentionally set the wrong path. So mostlikely the error is a better behavior here. NOTE: as far as I know, we are not tolerant to such user's input